### PR TITLE
[WSP-180] Search Component Improvements

### PR DESCRIPTION
**Pull request by GeoffreyHayward**
Switches over the SearchComponent to utilis a BR's EssentialListComponent to increase performance.

### CODE REVIEW

File: repository-data/application/src/main/resources/hcm-content/content/documents/corporate-website/faceted-search-by-date.yaml [hunk 1]
```diff
@@ -1,29 +1,34 @@
...
   - publicationsystem:GeographicCoverage
   - publicationsystem:Granularity
   - nationalindicatorlibrary:publishedBy
   - nationalindicatorlibrary:reportingLevel
   hippofacnav:limit: 100
+  hippofacnav:sortby:
+  - common:orderedSearchDate
+  - nationalindicatorlibrary:assuranceDate
+  - common:searchRank
+  - hippostdpubwf:lastModificationDate
```
Comment (davidbrrr) [inline repository-data/application/src/main/resources/hcm-content/content/documents/corporate-website/faceted-search-by-date.yaml:33]: Perhaps you should also specify the sortorder, as it defaults to ascending if not specified, but the component was using descending. See https://xmdocumentation.bloomreach.com/library/concepts/faceted-navigation/configure-sorting-and-limiting-of-result-sets.html

File: repository-data/application/src/main/resources/hcm-content/content/documents/corporate-website/faceted-search.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  - publicationsystem:Granularity
+  - nationalindicatorlibrary:publishedBy
+  - nationalindicatorlibrary:reportingLevel
+  hippofacnav:limit: 100
+  hippofacnav:sortby:
+  - common:searchRank
+  - jcr:score
+  - common:orderedSearchDate
+  - nationalindicatorlibrary:assuranceDate
+  - hippostdpubwf:lastModificationDate
```
Comment (davidbrrr) [inline repository-data/application/src/main/resources/hcm-content/content/documents/corporate-website/faceted-search.yaml:35]: Perhaps you should also specify the sortorder, as it defaults to ascending if not specified, but the component was using descending. See https://xmdocumentation.bloomreach.com/library/concepts/faceted-navigation/configure-sorting-and-limiting-of-result-sets.html

File: repository-data/webfiles/src/main/resources/site/freemarker/common/search-results-layout.ftl [hunk 3]
```diff
@@ -15,9 +14,8 @@
                     </div>
                 </div>
             </div>
-
             <div class="column column--two-thirds page-block page-block--main">
-                <@searchTabsComponent contentNames=hstResponseChildContentNames></@searchTabsComponent>
+                <@hst.include ref="tabs" />
```
Comment (davidbrrr) [inline repository-data/webfiles/src/main/resources/site/freemarker/common/search-results-layout.ftl:18]: Is there a reason for still including the tabs?

**Review by davidbrrr — COMMENTED** [review]
This seems like a bug, when I access the search I get no results in the list, and I have to select a facet for it to show results.
image

File: repository-data/application/src/main/resources/hcm-content/content/documents/corporate-website/faceted-search-by-date.yaml (hunk 1 above)
Comment (GeoffreyHayward) [reply repository-data/application/src/main/resources/hcm-content/content/documents/corporate-website/faceted-search-by-date.yaml:33]: Good spot. I will add that.

File: repository-data/application/src/main/resources/hcm-content/content/documents/corporate-website/faceted-search.yaml (hunk 2 above)
Comment (GeoffreyHayward) [reply repository-data/application/src/main/resources/hcm-content/content/documents/corporate-website/faceted-search.yaml:35]: Good spot. I will add that.

File: repository-data/webfiles/src/main/resources/site/freemarker/common/search-results-layout.ftl (hunk 3 above)
Comment (GeoffreyHayward) [reply repository-data/webfiles/src/main/resources/site/freemarker/common/search-results-layout.ftl:18]: I will remove this.

**Review by davidbrrr — CHANGES_REQUESTED** [review]
Finishing review.

**Review by davidbrrr — APPROVED** [review]

File: site/components/src/main/java/uk/nhs/digital/common/components/SearchComponent2.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    /*
+     * Copy of BR 14.7.13 doFacetedSearch so that "totalResults" can be captured.
+     */
+    protected <T extends EssentialsListComponentInfo> Pageable<HippoBean> doFacetedSearch(HstRequest request, T paramInfo, HippoBean scope) {
+        Pageable<HippoBean> pageable = DefaultPagination.emptyCollection();
+        String relPath = SiteUtils.relativePathFrom(scope, request.getRequestContext());
+        HippoFacetNavigationBean facetBean = ContentBeanUtils.getFacetNavigationBean(relPath, this.getSearchQuery(request));
+
+        if (facetBean != null) {
+            request.setAttribute("totalResults", facetBean.getCount());
```
[the remark this answers is not in the document]
Comment (GeoffreyHayward) [reply site/components/src/main/java/uk/nhs/digital/common/components/SearchComponent2.java:74]: Thanks, @davidbrrr; we used to have it set up this way, and it worked fine until we set the `resultset` limit on the facet. Once the limit was set the total results shown via `Pageable` was only in the range of 0 to 100 (i.e. the limit). 

Of course, when it's 100+ the user may need to use the filters to get to all of the results, but using `facetBean.getCount()` will always give the true size of the results.

File: site/components/src/main/java/uk/nhs/digital/common/components/SearchComponent2.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+            HippoResultSetBean resultSet = facetBean.getResultSet();
+            if (resultSet != null) {
+                HippoDocumentIterator<HippoBean> iterator = resultSet.getDocumentIterator(HippoBean.class);
+                pageable = this.getPageableFactory().createPageable(iterator, resultSet.getCount().intValue(), paramInfo.getPageSize(), this.getCurrentPage(request));
+            }
+        }
+
+        return (Pageable) pageable;
```
Comment (davidbrrr) [inline site/components/src/main/java/uk/nhs/digital/common/components/SearchComponent2.java:82]: Any reason for the cast to (Pageable) here? Seems extra and my IDE complained

**Pull request merged**